FRM_assm
==============

**Fastidious respiratory *Mycoplasma* whole genome assembly**

A *de novo* assembly pipeline for Fastidious respiratory *Mycoplasma spp.* using publicly available open source-tools. It takes in demultiplexed reads from GUPPY and putputs polished *de novo* assemblies. This pipleline is optomized for Mycoplasma spp. and can be run on most modern Mac or Linux computers. It is possible to use this pipeline with Apple Silicon Macs, either through Rosetta2 or by building the dependencies from source.



Manuals
-------

- [Installation instructions](docs/INSTALL.md)
- [Usage](docs/USAGE.md)

Publications
------------
*Pending review JCM*

Please also include the following references:

Mikhail Kolmogorov, Jeffrey Yuan, Yu Lin and Pavel Pevzner, "Assembly of Long Error-Prone Reads Using Repeat Graphs", Nature Biotechnology, 2019 https://doi.org/10.1038/s41587-019-0072-8

Li, H. (2021). New strategies to improve minimap2 alignment accuracy. Bioinformatics, 37:4572-4574. https://doi.org/10.1093/bioinformatics/btab705

Vaser, Sovic, I., Nagarajan, N., & Sikic, M. (2017). Fast and accurate de novo genome assembly from long uncorrected reads. Genome Research, 27(5), 737â€“746. https://doi.org/10.1101/gr.214270.116